### DISCUSSION TITLE
Tags or keywords

### DISCUSSION THREAD

**Discussion by hajo62** [bpatrik/pigallery2#533] [Q&A] [answered]
I've learned in #528 that tags should be used to specify the image used in the folder.
Tag/Keywords could be set with windows explorer or macs finder or other tools.

But: How does Pigallery recognize those metadata?

My workflow is 
1. Copy all my new photos to my computer
2. Synchronize (automatically) to my raspberry Pi (I use nextcloud to do this)
3. Sort and modify and delete some photos. Tag one image to be used as the image for the folder.
4. After each file modification I (re)set the last modification to the time the photo was taken (EXIF: _Aufnahmedatum_)
I have double checked, that the files synchronized to my Pi include the tag set 3.

When I than start a search in Pigallery (`keyword:"icon"`) I get only one image returned, although I have tagged more than one. Unfortunately I don't know, why at least this one is found; I assume, I made all modifications (and tagging) on a not synchronized folder and copy the file to a synched folder after all modifications...

I started manually (in settings) `Run now: Thumbnail Generation` as well as `Run now: Preview Filling` as well as `Indexing`.

**Comment by bpatrik** [comment, marked as answer]
The app should recognize if any photo in a folder is changed. But it only
recognizes it if you navigate to that folder or hit the indexing button.

Also in some system ppl reported that the app does not detect change. In
that case you can increase the indexing severity from low to medium or high.

-- Sorry for being brief, sent from my phone.

**Reply by hajo62** [reply]
Looks like modifying an embedded tag does not (always) reflect in a modified directory. Even not, when I change the file modification date of the tagged photo what I would expect. But adding and or removing files does, which I use as work around.

**Reply by kagahd** [reply]
Keep in mind that **changing** a rating of a photo may not be detected as a change because neither the file timestamp nor the file size have been changed by this operation. However, software programs as for example `rsync` or `WinSCP` rely on file timestamp and/or file size to detect changed files. 
The same applies when adding a tag while removing another tag of the same number of characters.
